controllers/configentry_controller.go

```diff
@@ -162,15 +168,17 @@ func (r *ConfigEntryController) ReconcileEntry(
 	}
 
 	if !configEntry.MatchesConsul(entry) {
-		_, _, err := r.ConsulClient.ConfigEntries().Set(configEntry.ToConsul(), &capi.WriteOptions{
+		logger.Info("config entry does not match consul", "modify-index", entry.GetModifyIndex())
+		_, writeMeta, err := r.ConsulClient.ConfigEntries().Set(configEntry.ToConsul(), &capi.WriteOptions{
 			Namespace: r.consulNamespace(req.Namespace),
 		})
 		if err != nil {
 			return r.syncUnknownWithError(ctx, logger, crdCtrl, configEntry, ConsulAgentError,
 				fmt.Errorf("updating config entry in consul: %w", err))
 		}
+		logger.Info("config entry updated", "request-time", writeMeta.RequestTime)
 		return r.syncSuccessful(ctx, crdCtrl, configEntry)
-	} else if configEntry.SyncedConditionStatus() == corev1.ConditionTrue {
+	} else if configEntry.SyncedConditionStatus() != corev1.ConditionTrue {
```

Comment:
This was a bug that was introduced during refactoring. I've fixed it here and added a test for it.